src/store/user/index.js

```diff
@@ -38,6 +66,37 @@ const mutations = {
 };
 
 const actions = {
+    async loginWithGoogle({ commit, state }) {
+        const gProvider = new GoogleAuthProvider();
+
+        try {
+            const res = await signInWithPopup(auth, gProvider);
+            // const credential = GoogleAuthProvider.credentialFromResult(res);
+            // const token = credential.accessToken;
+            const user = res.user;
+            commit('update-user', user);
+            commit('update-login-Modal', false);
+        } catch (err) {
+            // Handle Errors here.
+            const errorCode = error.code;
+            const errorMessage = error.message;
+            // The email of the user's account used.
+            const email = error.customData.email;
+            // The AuthCredential type that was used.
+            const credential = GoogleAuthProvider.credentialFromError(error);
+            console.log(errorCode, errorMessage, email, credential);
```

Comment:
do we need this log

Reply:
this is for development will remove when this feature is complete

Reply:
A good practice is to add a todo.
So that there is a track for the pending items.